src/logics/ArithLogic.cc

```diff
@@ -227,9 +238,15 @@ pair<PTRef, PTRef> ArithLogic::splitTermToVarAndConst(PTRef term) const {
         assert(getPterm(term).size() == 2);
         PTRef fac = getPterm(term)[0];
         PTRef var = getPterm(term)[1];
-        if (not isConstant(fac)) { std::swap(fac, var); }
+        if (not isConstant(fac)) {
+            std::swap(fac, var);
+            if (not isConstant(fac)) {
+                PTRef fac = yieldsSortInt(term) ? getTerm_IntOne() : getTerm_RealOne();
+                return {term, fac};
+            }
+        }
         assert(isConstant(fac));
-        assert(isNumVarLike(var));
+        assert(isNumVarLike(var) || isTimes(var));
```

Comment:
I think this case is going to be more difficult.
We have to think about how terms like `2*x*y` or `3*x*x` should be represented.

Reply:
Won't it work as Times(Times) in the current encoding?
Basically (* 3 (* x y))?
Take a look at the miniexample10.smt2 in the regression, it seems to work at least for some inputs like this


I'll think of it, ideally I would be able to come up with some example to see if what we have rn is not sufficiemt

Reply:
Then I think the encoding of Times needs to change. The terms should be normalized, same as for other operations.

Reply:
Should it be done in this PR though?
Seems like a separate issue tbh...

Reply:
This is a single issue of supporting nonlinear expression in`Logic`, so yes, should be done properly in one PR.

Reply:
Hmm, I looked into it again.

So in current PR we have Times on multiple arguments as well.

So (* 3 (* x y)) is actually (* x y 3)

But the thing is, do we want to return 3 as a constant here, or is it better to return 
{(* x y 3), 1} ?

Reply:
The second way would allow to preserve current code, while change would be more logical but would require much changes

Reply:
You definitely need to return {x*y, 3} in that case.

Reply:
It means that we need to create another multiplication inside of the function, but `splitTermToVarAndConst` is a constant one, and is expected to be const in a lot of functions

I can change it but not sure if it will be Ok

Reply:
Just to elaborate on it a bit, it will require to make `termSort` into not const for example

Reply:
I've added a special function to split multiplication, pls check it out...
Don't like the solution tbh, but it works...